src/client.rs

```diff
@@ -114,6 +117,20 @@ impl Client {
         }
     }
 
+    /// Get the response obtained when opening the session.
+    ///
+    /// Returns `None` if no session has yet been opened.
+    pub fn session_creation_response(&self) -> Option<&NewSessionResponse> {
```

Comment:
Ah, we can't actually directly return the type from `webdriver` here, as that would mean upgrading `webdriver` to a new major version would also require a new major version of `fantoccini`. This is what the [`wd` module](https://github.com/jonhoo/fantoccini/blob/main/src/wd.rs) is for — in there we have our own replicas of the `webdriver` types that are under _our_ control, and that we _can_ return. There, we can _also_ `#[derive(Clone)]` so that you won't need the `Arc` any more :tada: 

Just a note for when you add that type though, make sure to mark it as `#[non_exhaustive]` so that we can add fields to it in the future if the `webdriver` crate does!